src/share/doc/javatest/regtest/tag-spec.html

```diff
@@ -1249,6 +1249,12 @@ <h2 class="appendix" id="testvars">APPENDIX 2: TEST-SPECIFIC SYSTEM PROPERTIES a
 <td><code>TESTFILE</code>
 <td>The defining file of the test
 
+<tr>
+<td><code>test.main.class</code>
+<td><i>Not Available</i>
+<td>The computed name of the main entry-point of the current Java source file.
```

Comment:
Might be add notes that 
1) packages are supported, based on package line
2) inner classes are not supported
3) the test fails during execution if ${test.main.class} is not found or doesn't have main() method. 
It is just a suggestion, please update as you think it would be better. No need to overload users with info.

Reply:
The error message produced when the executable can't be found should be very helpful.
Not specific to .java or the smart tag though.

On a similar typo, I got.
`TEST RESULT: Error. can't find FilePermissionTestX in test directory or libraries`
If it is a class name, it would be useful to say its a "class".

Reply:
Refined the documentation. A more helpful error message would require a deeper knowledge of the setup at the time of the name computation. It is up to the test author to either specify a hard-coded fully-qualified class name as of now; or ensure the heuristic based on the file name is matched.